### ISSUE TITLE
Provide WiFi AP and Serial Menu function to open a new log file with new filename based on current time

### ISSUE THREAD

**Issue by tonycanike** [sparkfun/SparkFun_RTK_Firmware#344]
### Subject of the issue
Provide the user with the ability to open a clean fresh log file after configuring the rates, messages, etc.

The user should be able to correlate the log file datetime in the filename with the start of their logging session.  This is important when post-processing multiple log files from multiple occupations.

Currently the power-on time is used to name the log file.  The power-on time is not the same as the time the logging session actually started, as the user may need to configure the logging with the WiFi AP.

Also see #341 and #346.

### Your workbench
Facet (the version before the nice cases, someday I'll write down the product number :) 
Firmware 2.5 and also the 10/1/2022 RC version.

### Steps to reproduce & Actual Behavior
1.  User sets up Facet on tripod over a point.
2. User turns on Facet.   (Facet automatically begins logging to files.)
3. User begins WiFI AP configuration.
4. User configures rates, messages, and logging as necessary for the project.
6. User saves configuration and exit.  Facet restarts.
7. NOTE:  This moment is the start of the Static Logging Session the user desires.  User writes down the UTC start time of the logging session.
8. Facet unfortunately reopens old log file (sometimes very old even when newer files exist, but that's issue #343 )
9. Facet starts logging to old file.
10. Log file has messages from before it was properly configured.   
11. Log file possibly has corrupted messages from when the user was configuring the Facet via AP and consuming system resources.
12. Four hours later user turns off Facet
13. Log file names don't match the time of the actual logging session starttime.  User gets confused.
14. Much wrangling with SD cards and cables ensues (#323 would help.)
15. UBX_Integrity_checker.py complains about many resyncs in the first 10-15 minutes of the UBX file.  (there are others hours into the session too, but that's issue #345 .)
16. Much wrangling with RTKCONV ensues.
17. User discovers old, even very old, log messages in file.
17. User spends much time with RTKCONV to try to delete old messages in the log.  Meanwhile his friends with Sokkia and Carlson GNSS receivers have long-since finished sending their data to OPUS and they laugh at me while drinking their beers. :)
19. RINEX file submitted to OPUS.
20. OPUS complains loudly.

### Expected behavior 

Ideally, the user would be able to change logging configuration, close and open log files, create new log files, etc., without rebooting the Facet.

Ideally, the user would be able to connect to the WiFI AP configuration page to check the progress of the logging (approx location, log file name, message rates, messages configured, log file size, SD free space, signal strengths, satellites, etc.) without disrupting the logging.

We don't always get the everything we want, and the Facet is an amazing device at the price point and we totally know it.

So here's a use case:

**Sample Sufficient Use Case:   Static PPP Logging**
1.  User sets up Facet on tripod over a point.
2. User turns on Facet.   (Facet automatically begins logging to files.)
3. User begins WiFI AP configuration.
4. User configures rates, messages, and logging as necessary for the project.
6. User saves configuration.
7. User re-enters configuration menu, verifies rates, messages, etc.
8. User Exits and Resets.  User writes down the UTC time. Facet restarts.
9. Facet reopens **NEW** log file and starts logging.  Log filename is within a minute of the time written down in the above step.
10. Log file doesn't have any messages from before it was properly configured and restarted
11. Four hours later user turns off Facet
12. Log filename datetime matches (+/- one minutes)  the UTC time of the start of the logging session.
13. Much wrangling with SD cards, cables, RTKCONV, and file management occurs.  (gosh these are separate topics from this issue.)
14.  UBX_Integrity_checker.py finds no resyncs.
15. RINEX file submitted to OPUS.
16. OPUS provides accurate solution.

**Comment by nseidle** [comment]
I thoroughly enjoy these issues and appreciate you separating them! I'm sorry your other friends get to beer cracking before you! Let's fix that.

Your report is long but the title (I believe) summarizes things nicely: 'get me a log reset button on the AP page (and serial menu)'. Let me know if that's incorrect.

**Comment by nseidle** [comment]
![image](https://user-images.githubusercontent.com/117102/194946154-b06a3842-1406-4383-b4b9-df75db2bbe81.png)

Added

**Comment by nseidle** [comment]
![image](https://user-images.githubusercontent.com/117102/194946272-e73cea9d-4a8f-4cf8-82ce-7e7e461d1a6f.png)

Added serial menu

**Comment by tonycanike** [comment]
Awesome, thank you very much!

**Comment by amlago** [comment]
This looks great.
What remains after is to be able to copy these files without removing the sd by cable or another method.
Good work.

**Comment by tonycanike** [comment]
@nseidle is there any way we could get feedback after we push the "Start New Log" button in the WiFi config?  

I finally got around to trying it out (testing the logging fixes in the 26-Oct build this morning).  I think I pushed the button but not sure it did anything.  Maybe display the new logging file name, or just a message that a new log file will be created after reboot, etc.  I'm not exactly sure if pushing the button immediately closes the old and opens a new file, or if that happens after reboot.

If the logging to the SD card is still running 'in the background' when WiFi AP config is active, then it'd be really cool to see the live log file name.  (I'll open a separate issue if you want.)  And the filename could change when someone presses the "Start New Log" button.
